internal/semver/semver_test.go

```diff
@@ -218,71 +220,70 @@ func TestCanUpgrade(t *testing.T) {
 	testCases := map[string]struct {
 		version1 Semver
 		version2 Semver
-		want     bool
 		wantErr  bool
```

Comment:
wantErr is misleading naming. the orignal `want` was used to indicate if we accept an upgrade. 
the assertion actually checks the reverse of what the naming suggests:
`			assert.Equal(tc.wantErr, tc.version2.IsUpgradeTo(tc.version1) == nil)`
i.e. wantErr needs to be true when it has not error.

I'd rename to `wantUpgrade`.